server/sonar-server/src/main/java/org/sonar/server/computation/metric/MetricRepositoryImpl.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.metric;
+
+import org.sonar.core.metric.db.MetricDto;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.server.db.DbClient;
+import org.sonar.server.metric.persistence.MetricDao;
+
+import static java.util.Objects.requireNonNull;
+
+public class MetricRepositoryImpl implements MetricRepository {
+  private final DbClient dbClient;
+  private final MetricDao metricDao;
+
+  public MetricRepositoryImpl(DbClient dbClient, MetricDao metricDao) {
+    this.dbClient = dbClient;
+    this.metricDao = metricDao;
+  }
+
+  @Override
+  public Metric getByKey(String key) {
+    requireNonNull(key);
+
+    try (DbSession dbSession = dbClient.openSession(false)) {
```

Comment:
I haven't made any assumption on how this should be optimized yet since I don't know how it is used (only use I have yet can really work with this simple implementation)